api/internal/localizer/localizer_test.go

```diff
@@ -299,3 +299,160 @@ patches:
+  kind: ConfigMapGenerator
+  literals:
+  - APPLE=orange
+  metadata:
+    name: another-map
+  ---
+  apiVersion: builtin
+  kind: SecretGenerator
+  literals:
+  - APPLE=b3Jhbmdl
+  metadata:
+    name: secret
+  options:
+    disableNameSuffixHash: true
+kind: Kustomization
+`,
+		"plugin.yaml": `apiVersion: builtin
+kind: ConfigMapGenerator
+metadata:
+  name: map
+`,
+	}
+	addFiles(t, fSys, "/a", kustAndPlugins)
+
+	lclzr := createLocalizer(t, fSys, "/a", "", "/alpha/dst")
+	require.NoError(t, lclzr.Localize())
+
+	fSysExpected := makeMemoryFs(t)
+	addFiles(t, fSysExpected, "/a", kustAndPlugins)
+	addFiles(t, fSysExpected, "/alpha/dst", map[string]string{
+		"kustomization.yaml": kustAndPlugins["kustomization.yaml"],
+	})
+	checkFSys(t, fSysExpected, fSys)
+}
+
+func TestLocalizeTransformers(t *testing.T) {
+	fSys := makeMemoryFs(t)
+	kustAndPlugins := map[string]string{
+		"kustomization.yaml": `apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+transformers:
+- |
+  apiVersion: builtin
+  jsonOp: '[{"op": "add", "path": "/spec/template/spec/dnsPolicy", "value": "ClusterFirst"}]'
+  kind: PatchJson6902Transformer
+  metadata:
+    name: patch6902
+  target:
+    name: deployment
+  ---
+  apiVersion: builtin
+  kind: ReplacementTransformer
+  metadata:
+    name: replacement
+  replacements:
+  - source:
+      fieldPath: spec.template.spec.containers.0.image
+      kind: Deployment
+    targets:
+    - fieldPaths:
+      - spec.template.spec.containers.1.image
+      select:
+        kind: Deployment
+- plugin.yaml
+`,
+		"plugin.yaml": `apiVersion: builtin
+kind: PatchStrategicMergeTransformer
+metadata:
+  name: patchSM
+paths:
+- pod.yaml
+`,
+	}
+	addFiles(t, fSys, "/a", kustAndPlugins)
+
+	lclzr := createLocalizer(t, fSys, "/a", "", "/dst")
+	require.NoError(t, lclzr.Localize())
+
+	fSysExpected := makeMemoryFs(t)
+	addFiles(t, fSysExpected, "/a", kustAndPlugins)
+	addFiles(t, fSysExpected, "/dst", map[string]string{
+		"kustomization.yaml": kustAndPlugins["kustomization.yaml"],
+	})
+	checkFSys(t, fSysExpected, fSys)
+}
+
+func TestLocalizeBuiltinPluginsInvalid(t *testing.T) {
+	type testCase struct {
+		name  string
+		files map[string]string
+	}
+	for _, test := range []testCase{
+		{
+			name: "bad_inline_resource",
+			files: map[string]string{
+				"kustomization.yaml": `apiVersion: kustomize.config.k8s.io/v1beta1
+generators:
+- |
+  apiVersion: builtin
+  kind: ConfigMapGenerator
+kind: Kustomization
+`,
+			},
+		},
+		{
+			name: "bad_file_resource",
+			files: map[string]string{
+				"kustomization.yaml": `apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+transformers:
+- plugin.yaml
+`,
+				"plugin.yaml": `apiVersion: builtin
+metadata:
+  name: PatchTransformer
+`,
+			},
+		},
+	} {
+		t.Run(test.name, func(t *testing.T) {
+			fSys := makeMemoryFs(t)
+			addFiles(t, fSys, "/", test.files)
+			lclzr := createLocalizer(t, fSys, "/", "", "/dst")
+			require.Error(t, lclzr.Localize())
```

Comment:
For error cases, we should also check the contents of the error message itself, not just that there is an error. We don't want to be throwing the wrong error, and this will also help us to verify that the format of the error is correct without any weird spacing or extra characters.

```
actualErr := lclzr.Localize()
require.Error(t, actualErr)
require.Equal(t, actualErr, test.expectedErr)
```

Similar comment for `TestLocalizeValidators`